src/qibo/config.py

```diff
@@ -37,9 +37,21 @@ def raise_error(exception, message=None, args=None):
     os.environ["TF_CPP_MIN_LOG_LEVEL"] = str(LOG_LEVEL)
     import numpy as np
     import tensorflow as tf
+
     # Backend access
     K = tf
 
+    # Set the intra number of threads to the environment flag
+    if "QIBO_NUM_THREADS" in os.environ:
+        nthreads = int(os.environ["QIBO_NUM_THREADS"])
+        tf.config.threading.set_intra_op_parallelism_threads(nthreads)
+
+
+    THREADS = {
+        'inter': tf.config.threading.get_inter_op_parallelism_threads(),
+        'intra': tf.config.threading.get_intra_op_parallelism_threads()
+    }
```

Comment:
If you think that this functionality would be useful in other applications apart from `optimparallel` (which is not very interesting case sinc the NUM_THREADS value is constrained to be 1), we could possibly add `set_threads`/`get_threads` methods similar to the `set_device`/`get_device` functionality.

Reply:
Unfortunately we can't change the inter/intra after allocating tf objects, this should be the first thing to do after importing tf. However we could add to the custom tf operators an argument with the number of allowed cores for the thread pooling objects (but this will only for our operators). 

This is connected to your previous comment about changing automatically the number of threads to 1.